system/jlib/jlog.hpp

```diff
@@ -1342,4 +1348,301 @@ interface IComponentLogFileCreator : extends IInterface
+        limit = l.limit;
+        timeRange = l.timeRange;
+        filter = l.filter;
+        offset = l.offset;
+
+        return *this;
+    }
+
+    ~LogAccessConditions()
+    {
+        logFieldNames.kill(true);
+    }
+
+    void appendLogFieldName(const char * fieldname)
+    {
+        logFieldNames.append(fieldname);
+    }
+
+    void copyLogFieldNames(const StringArray * fields)
+    {
+        if (*fields)
+        {
+            ForEachItemIn(fieldsindex,*fields)
+            {
+                logFieldNames.append(fields->item(fieldsindex));
+            }
+        }
+    }
+    inline const StringArray &queryLogFieldNames() { return logFieldNames; }
+};
+
+interface IRemoteLogAccess : extends IInterface
+{
+    virtual bool fetchLog(LogAccessConditions & options, StringBuffer & returnbuf) = 0;
+
+    virtual const char * getRemoteLogAccessType() = 0;
+    virtual IPropertyTree * fetchLogMap() = 0;
+    virtual const char * fetchConnectionStr() = 0;
+};
+
+inline bool fetchLog(StringBuffer & returnbuf, IRemoteLogAccess & logAccess, ILogAccessFilter * filter, LogAccessTimeRange timeRange, StringArray & cols)
+{
+    LogAccessConditions logFetchoptions;
+    logFetchoptions.timeRange = timeRange;
+    logFetchoptions.filter = filter;
+    logFetchoptions.copyLogFieldNames(&cols); //ensure these fields are declared in m_logMapping->queryProp("WorkUnits/@contentcolumn")? or in LogMap/Fields?"
+
+    return logAccess.fetchLog(logFetchoptions, returnbuf);
+}
+
+inline bool fetchJobIDLog(StringBuffer & returnbuf, IRemoteLogAccess & logAccess, const char *jobid, LogAccessTimeRange timeRange, StringArray & cols)
+{
+    return fetchLog(returnbuf, logAccess, getWUIDLogAccessFilter(jobid), timeRange, cols);
+}
+
+inline bool fetchComponentLog(StringBuffer & returnbuf, IRemoteLogAccess & logAccess, const char * component, LogAccessTimeRange timeRange, StringArray & cols)
+{
+    return fetchLog(returnbuf, logAccess, getComponentLogAccessFilter(component), timeRange, cols);
+}
+
+inline bool fetchLogByAudience(StringBuffer & returnbuf, IRemoteLogAccess & logAccess, MessageAudience audience, LogAccessTimeRange timeRange, StringArray & cols)
+{
+    return fetchLog(returnbuf, logAccess, getAudienceLogAccessFilter(audience), timeRange, cols);
+}
+
+inline bool fetchLogByClass(StringBuffer & returnbuf, IRemoteLogAccess & logAccess, LogMsgClass logclass, LogAccessTimeRange timeRange, StringArray & cols)
+{
+    return fetchLog(returnbuf, logAccess, getClassLogAccessFilter(logclass), timeRange, cols);
+}
+
+//logAccessPluginConfig expected to contain connectivity and log mapping information
+typedef IRemoteLogAccess * (*newLogAccessPluginMethod_t_)(IPropertyTree & logAccessPluginConfig);
+
+class jlib_decl RemoteLogAccessLoader
+{
+public:
+    RemoteLogAccessLoader() = default;
+    ~RemoteLogAccessLoader();
+
+    template <class RemoteLogAccessPlugin>
+    static RemoteLogAccessPlugin * queryRemoteLogAccessor()
```

Comment:
how should the framework work, if there are multiple log sinks (is that possible)?
Might we want to have multiple plugins used from different contexts?

Reply:
I wasn't able to develop a scenario where we'd direct our logs to multiple targets. If we ever encounter a need for multiple targets, I'd imagine it'd be done at the component level. This function could accept the configuration supplied by the caller instead of hardcoding the global config...

Reply:
Only change here is the component yaml is queried, rather than global.
yaml template work likely needed to copy global logaccess to component.